packages/site-content/src/types/eleventy.d.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+declare module '@11ty/eleventy' {
+  export interface EleventyConfig {
```

Comment:
eleventy calls this class the `UserConfig`: https://unpkg.com/browse/@11ty/eleventy@1.0.2/src/UserConfig.js and they say you can get come typings with this:

```
/** @param {import("@11ty/eleventy").UserConfig} eleventyConfig */
module.exports = function (eleventyConfig) {
  // …
};
```
https://www.11ty.dev/docs/config/#type-definitions

Is it possible to just augment what types TS can infer on that import (if it supports a type of a JS module at all? I'm not sure), and should we align the name of the interface?

Also, here's what seems like a pretty nice declaration file if we want to cross check our definitions or import some, licenses permitting (it's MIT): https://github.com/chriskrycho/v5.chriskrycho.com/blob/main/types/eleventy.d.ts

Reply:
Adding the `@param` comment doesn't do anything at all for me. It just gets implicit `any`.

https://unpkg.com/browse/@11ty/eleventy@1.0.2/src/UserConfig.js  defines things like `addPlugin` on an interface called `Config`, and then there's a thing called `UserConfig` which strips off the methods.

So are you sure that's the thing that's passed as the `eleventyConfig` parameter, or is it supposed to be the thing you return with `dir` etc.?

I'm sort of confused and the name doesn't seem to matter very much, so I'll just leave things as they are for now.

Reply:
Huh... the location of that `@param` annotation says that it should be the type of the param, not the return. Given the view of types on the project, I wouldn't be surprised if that setup wasn't tested very well.